applications/GeoMechanicsApplication/custom_processes/reset_displacement_process.cpp

```diff
@@ -0,0 +1,41 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#include "reset_displacement_process.h"
+#include "includes/mat_variables.h"
+#include "includes/model_part.h"
+#include "includes/ublas_interface.h"
+#include "processes/set_initial_state_process.h"
+
+namespace Kratos
+{
+
+ResetDisplacementProcess::ResetDisplacementProcess(ModelPart& rModelPart, const Parameters&)
+    : mrModelPart(rModelPart)
+{
+}
+
+void ResetDisplacementProcess::ExecuteInitializeSolutionStep()
+{
+    SetInitialStateProcess<3> set_initial_state_process(mrModelPart);
+    set_initial_state_process.ExecuteInitializeSolutionStep();
```

Comment:
That's a good question, for the sake of trying out this process, I just forwarded the call (since the function on the process itself is implemented on the `ExecuteInitializeSolutionStep`. However, I don't really see a reason for having to set the initial state each time (since the constitutive laws stay alive during an entire stage)